wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserService.java

```diff
@@ -391,84 +286,30 @@ public UserInRoleDescriptor inRole(@Required @Description("role inside a scope")
                                                  .withScopeId(scopeId);
     }
 
-    /**
-     * Get user by name.
-     *
-     * @param name
-     *         user name
-     * @return found user
-     * @throws NotFoundException
-     *         when user with given name doesn't exist
-     * @throws ServerException
-     *         when some error occurred while retrieving user
-     */
-    @GET
-    @Path("/name/{name}")
-    @GenerateLink(rel = "get user by name")
-    @RolesAllowed({"user", "system/admin", "system/manager"})
-    @Produces(APPLICATION_JSON)
-    @ApiOperation(value = "Get user by name",
-                  notes = "Get user by its name in the system. Roles allowed: user, system/admin, system/manager.")
-    @ApiResponses({@ApiResponse(code = 200, message = "OK"),
-                   @ApiResponse(code = 404, message = "Not Found"),
-                   @ApiResponse(code = 500, message = "Internal Server Error")})
-    public UserDescriptor getByName(@ApiParam(value = "User email")
-                                    @PathParam("name")
-                                    String name) throws NotFoundException, ServerException {
-        final User user = userManager.getByName(name);
-        return injectLinks(toDescriptor(user), getServiceContext());
-    }
-
-    /**
-     * Get setting of user service
-     */
-    @GET
-    @Path("/settings")
-    @Produces(APPLICATION_JSON)
-    public Map<String, String> getSettings() {
-        return ImmutableMap.of(USER_SELF_CREATION_ALLOWED, Boolean.toString(userSelfCreationAllowed));
+    private User fromToken(String token) throws UnauthorizedException, ConflictException {
+        final String email = tokenValidator.validateToken(token);
+        final int atIdx = email.indexOf('@');
+        // Getting all the characters before '@' e.g. user@codenvy.com -> user
```

Comment:
as I said previously are the checks around token/user should not be handled by class managing token and users ?